TKGame/Players/Components/C_Player_WeaponAttack.cs

```diff
@@ -0,0 +1,98 @@
+﻿using Microsoft.Xna.Framework.Graphics;
+using Microsoft.Xna.Framework.Input;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+using TKGame.BackEnd;
+using TKGame.Components.Interface;
+
+namespace TKGame.Players.Components
+{
+    internal class C_Player_WeaponAttack : IGraphicsComponent
+    {
+        #region Variables
+        private int swordX = 40;
+        private int swordY = 10;
+        private int spearX = 100;
+        private int spearY = 10;
+        private int axeX = 40;
+        private int axeY = 10;
+        #endregion
+        public ComponentType Type => throw new NotImplementedException();
+
+        public void Update(Entity player)
+        {
+
+            if (Input.MouseState.LeftButton == ButtonState.Pressed) //Checks if left mouse is pressed
+            {
+                if (player.weapon.isActiveSword)//Checks if weapon is a sword
+                    SwordAttack(player);
+                if(player.weapon.isActiveSpear)//Checks if weapon is a spear
+                    SpearAttack(player);
+                if(player.weapon.isActiveAxe)// cheacks if weapon is an axe
+                    AxeAttack(player);
```

Comment:
I feel like there should be a better way to determine the weapon that should be used. Player should have a weapon component right?